api/operator/v1alpha1/certmanager_types.go

```diff
@@ -59,6 +60,50 @@ type CertManagerSpec struct {
 	// +kubebuilder:validation:Optional
 	// +optional
 	CAInjectorConfig *DeploymentConfig `json:"cainjectorConfig,omitempty"`
+
+	// DefaultNetworkPolicy enables the default network policy for cert-manager components.
+	// When set to "true", the operator will create default network policies to secure
+	// communication between cert-manager controller, webhook, and cainjector components.
+	// When set to "false" or empty, no default network policies are created.
+	// Valid values are: "true", "false", or empty (default: false).
+	//
+	// This field is immutable once set to "true" for security reasons. Network policies
+	// cannot be disabled once enabled to prevent accidental security degradation.
+	// Users should carefully plan their network policy requirements before enabling this field.
```

Comment:
Users should have a way to disable it once enabled. Operator doesn't need to handle the NP resources cleanup, but if it's an irreversible operation, it would leave no room for users to take any mitigation actions once unforeseen issue arisen.